test/unit/puppet_fact_parser_test.rb

```diff
@@ -233,8 +234,35 @@ def setup
     assert_equal '172.30.43.0', parser.interfaces['local_area_connection_2'][:network]
   end
 
+  test "should create connection between new os and new arch on facts parsing" do
+    assert os_to_arch_connectivity_check
+  end
+
+  test "should create connection between existing os and new arch on facts parsing" do
+    FactoryGirl.create(:operatingsystem,:name => 'AnyOS', :major => '6')
+    assert os_to_arch_connectivity_check
+  end
+
+  test "should create connection between existing os and existing arch on facts parsing" do
+    FactoryGirl.create(:operatingsystem,:name => 'AnyOS', :major => '6')
+    FactoryGirl.create(:architecture,:name => 'AnyArc')
+    assert os_to_arch_connectivity_check
+  end
+
+  test "should create connection between new os and existing arch on facts parsing" do
+    FactoryGirl.create(:architecture,:name => 'AnyArc')
+    assert os_to_arch_connectivity_check
+  end
+
   private
 
+  def os_to_arch_connectivity_check
+    @importer = PuppetFactParser.new({'operatingsystem' => 'AnyOS', 'architecture' => 'AnyArc', 'operatingsystemrelease' => '6'})
+    os = importer.operatingsystem
+    arch = importer.architecture
+    arch.operatingsystems.include?(os) && os.architectures.include?(arch)
```

Comment:
I covered this in my example refactoring comment above, but only one of these checks should be necessary.